cl/persistence/state/state_accessors.go

```diff
@@ -43,6 +76,35 @@ func IncrementPublicKeyTable(tx kv.RwTx, state *state.CachingBeaconState, prever
 	return nil
 }
 
+func IncrementHistoricalRootsTable(tx kv.RwTx, state *state.CachingBeaconState, preverifiedIndicies uint64) error {
+	for i := preverifiedIndicies; i < state.HistoricalRootsLength(); i++ {
+		key := base_encoding.Encode64ToBytes4(i)
+		root := state.HistoricalRoot(int(i))
+		if err := tx.Put(kv.HistoricalRoots, key, root[:]); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func IncrementHistoricalSummariesTable(tx kv.RwTx, state *state.CachingBeaconState, preverifiedIndicies uint64) error {
+	var temp []byte
+	var err error
+	for i := preverifiedIndicies; i < state.HistoricalSummariesLength(); i++ {
```

Comment:
can't we do a size estimation for temp at the start here? or is temp tiny and it doesnt matter

Reply:
temp is 80 bytes and will be reused